handlers.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"encoding/json"
+	"github.com/gorilla/mux"
+	"gopkg.in/mgo.v2/bson"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+func Auth(fn http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		// TODO Handle auth.
+		// Only continue if user is authorized
+		fn(w, r)
+	}
+}
+
+func (sf *Sunfish) AddFile(w http.ResponseWriter, r *http.Request) {
+	// Handles a Post Request for a Sia file and saves it to the DB
+	var siafile Siafile
+	body, err := ioutil.ReadAll(io.LimitReader(r.Body, 1048576))
+
+	if err != nil {
+		panic(err)
+	}
+
+	if err := r.Body.Close(); err != nil {
+		panic(err)
+	}
+
+	if err := json.Unmarshal(body, &siafile); err != nil {
+		w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+		w.WriteHeader(422) // unprocessable entity
+		if err := json.NewEncoder(w).Encode(err); err != nil {
+			panic(err)
+		}
```

Comment:
in siad we would simply write:
```go
http.Error(w, "Failed to decode sia file: "+err.Error(), http.StatusBadRequest)
```
Is there a reason why the error should be encoded as JSON?